### DISCUSSION TITLE
Multicore Image Layout

### DISCUSSION THREAD

**Discussion by danielb-tkjcy3rq** [zephyrproject-rtos/zephyr#69769] [Q&A]
Hi All,

Probably a dumb question but I can't seem to find the answer on my own. I'm running the openamp sample (zephyr v3.5.0) on NXP's mimxrt1160 evk,  which is all working fine. The thing I don't understand is what the layout of the images on flash is. After the build, is there just one image that combines the host and remote images and that would get flashed in one go? Or are the two images that are flashed separately?  Where in flash are these images being written to?

**Comment by danielb-tkjcy3rq** [comment]
I think I understand this a bit more now. The CM7 images is being written to flash in slot0_partition, while the CM4 image is in slot1_partition. Am I right in thinking that the openamp imx1160_evk sample wouldn't work with mcuboot because slot1_partition is needed as a secondary slot?

Does anoyone know if there is a way of of merging the two images (CM7 and CM4) into one binary that is then contained in one partition in flash? So the CM4 image would be at a certain offset from the end of the CM7 image. This would be ideal for my project

Thanks,
Daniel

**Comment by DerekSnell** [comment]
Hi @danielb-tkjcy3rq ,
Yes, in the RT1160, the chosen node `nxp,m4-partition` is where the CM4 image is stored in flash.  By default, the board file mimxrt1160_evk_mimxrt1166_cm4.dts places that node in [slot1_partition](https://github.com/zephyrproject-rtos/zephyr/blob/8de8bcc99c486ccb05faed39869773230be32fa0/boards/nxp/mimxrt1160_evk/mimxrt1160_evk_mimxrt1166_cm4.dts#L28).

Yes, MCUboot typically uses the slot1 partition for firmware image update.  See our other discussion about this in https://github.com/zephyrproject-rtos/zephyr/discussions/70448.

There are utilities available to combine multiple images into a single image file.  I found [this one](https://developer.arm.com/documentation/ka004500/latest/) doing an online search.

Best regards

**Comment by danielb-tkjcy3rq** [comment]
Thank you @DerekSnell.
